src/Microsoft.Diagnostics.EventFlow.Inputs.Etw/StandardTraceEventSession.cs

```diff
@@ -20,15 +20,46 @@ internal class StandardTraceEventSession : ITraceEventSession
         private bool isProcessing;
         private IHealthReporter healthReporter;
 
-        public StandardTraceEventSession(IHealthReporter healthReporter)
+        public StandardTraceEventSession(string sessionNamePrefix, bool cleanupOldSessions, bool restartExisting, IHealthReporter healthReporter)
         {
             Requires.NotNull(healthReporter, nameof(healthReporter));
+            Requires.NotNullOrWhiteSpace(sessionNamePrefix, nameof(sessionNamePrefix));
 
             this.healthReporter = healthReporter;
-            this.inner = new TraceEventSession($"EventFlow-{nameof(EtwInput)}-{Guid.NewGuid().ToString()}", TraceEventSessionOptions.Create);
+
+            string sessionName = null;
+            if (cleanupOldSessions)
+            {
+                sessionName = CleanupMatchingSessions(sessionNamePrefix, restartExisting);
+            }
+            if (sessionName == null)
+                sessionName = $"{sessionNamePrefix }-{Guid.NewGuid().ToString()}";
+
+            // even if the session already exists, we must restart it as we cannot enable providers on an attached session
+            this.inner = new TraceEventSession(sessionName, TraceEventSessionOptions.Create);
             this.isProcessing = false;
         }
 
+        string CleanupMatchingSessions(string sessionNamePrefix, bool keepOne)
+        {
+            string result = null;
+
+            foreach (var sesName in TraceEventSession.GetActiveSessionNames())
```

Comment:
I do not see the `keepOne` parameter being used--looks like a session is always reused?

Please rename the parameter to `reuseExistingSession` to keep it consistent with configuration